packages/augur-core/source/contracts/libraries/token/ERC20.sol

```diff
@@ -0,0 +1,227 @@
+pragma solidity ^0.5.0;
+
+import 'ROOT/libraries/token/IERC20.sol';
+import 'ROOT/libraries/math/SafeMathUint256.sol';
+
+
+/**
+ * @dev Implementation of the {IERC20} interface.
+ *
+ * This implementation is agnostic to the way tokens are created. This means
+ * that a supply mechanism has to be added in a derived contract using {_mint}.
+ * For a generic mechanism see {ERC20Mintable}.
+ *
+ * TIP: For a detailed writeup see our guide
+ * https://forum.zeppelin.solutions/t/how-to-implement-erc20-supply-mechanisms/226[How
+ * to implement supply mechanisms].
+ *
+ * We have followed general OpenZeppelin guidelines: functions revert instead
+ * of returning `false` on failure. This behavior is nonetheless conventional
+ * and does not conflict with the expectations of ERC20 applications.
+ *
+ * Additionally, an {Approval} event is emitted on calls to {transferFrom}.
+ * This allows applications to reconstruct the allowance for all accounts just
+ * by listening to said events. Other implementations of the EIP may not emit
+ * these events, as it isn't required by the specification.
+ *
+ * Finally, the non-standard {decreaseAllowance} and {increaseAllowance}
+ * functions have been added to mitigate the well-known issues around setting
+ * allowances. See {IERC20-approve}.
+ */
+contract ERC20 is IERC20 {
```

Comment:
Was this pulled from OpenZeppelin or somewhere else?  The prolific comments suggests that it wasn't hand-rolled in-house.  😉

Reply:
Ah, looking at the comments it appears OZ.  Might be worth including a permalink to the OZ contract.